xwiki-platform-core/xwiki-platform-flamingo/xwiki-platform-flamingo-skin/xwiki-platform-flamingo-skin-resources/src/main/resources/flamingo/docextra.vm

```diff
@@ -155,8 +155,11 @@
         ## Override keyboard shortcut (if any)
         ##
         #if ($keyboardShortcutsEnabled && "$!extraShortcut" != "")
-          shortcut.remove("$extraShortcut");
-          shortcut.add("$extraShortcut", function() { XWiki.displayDocExtra("${extraAnchor}", "${extraTemplate}", true); }, { 'disable_in_input':true });
+          #set ($userPreferenceName = "shortcut_view_${extraName}")
+          #set ($shortcut_view_extra = $!{xwiki.getUserPreference($userPreferenceName)})
+          #set ($extraShortcutWithUserPreferences = "#if($shortcut_view_extra!='')$shortcut_view_extra#{else}$extraShortcut#end")
+          shortcut.remove("$extraShortcutWithUserPreferences");
+          shortcut.add("$extraShortcutWithUserPreferences", function() { XWiki.displayDocExtra("${extraAnchor}", "${extraTemplate}", true); }, { 'disable_in_input':true });
```

Comment:
I'm not sure what kind of escaping is necessary here. There's definitely some needed escaping since the userPreference value is something any user types in their profile. However the `getShortcutValue` macro itself is velocity templating, which is used inside a js file (actionButtons.js), a js script block in some html template (shortcuts.vm), and a html template (editActionButton > submit button > name attribute of an input).
The context is varied so I'm not sure what escaping would be appropriate...
Should we just add two escaping, XML and javascript ones?

Reply:
I added both escaping to the result of the velocity macro, better safe than sorry.

Reply:
Please don't just escape twice as escaping causes real bugs. The best practice for escaping is to put it where you put the output into the context which then also tells you what escaping is needed.

Reply:
Addressed in https://github.com/xwiki/xwiki-platform/pull/2981/commits/74f90d2e6fb8da0fee19b1d1256e637f135e9b87 :+1: